src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/SwitchHelper.groovy

```diff
@@ -164,13 +190,13 @@ class SwitchHelper {
         Wrappers.wait(Constants.RULES_INSTALLATION_TIME) {
             def actualHexCookie = []
             for (long cookie : northbound.getSwitchRules(sw.dpId).flowEntries*.cookie) {
-                actualHexCookie.add(Cookie.decode(cookie).toString())
+                actualHexCookie.add(decode(cookie).toString())
             }
             def expectedHexCookie = []
             for (long cookie : sw.defaultCookies) {
-                expectedHexCookie.add(Cookie.decode(cookie).toString())
+                expectedHexCookie.add(decode(cookie).toString())
             }
-            assert actualHexCookie.sort() == expectedHexCookie.sort()
+            assertThat actualHexCookie, containsInAnyOrder(expectedHexCookie.toArray())
```

Comment:
Will be false positive in next case:
1. switch is in multiTable mode
2. change switch mode to single table
3. For some reasons all rules were **NOT** reinstalled.

**Expected result:**
Assertion failed

**Actual result**
Assert is OK because:

Single table rules are a subset of multi table rules (see line 143)

so in our case actualHexCookie = [singleTableRules + excess multi table rules]
expectedHexCookies = [singleTableRules]

that is why `actualHexCookie. containsInAnyOrder(expectedHexCookie.toArray())` is OK

Reply:
Even though the matcher is called 'contains..', it does a full match and will complain in both cases if actual is bigger than expected or expected is bigger than actual.

Reply:
okay